ee/tabby-webserver/src/service/license.rs

```diff
@@ -1,20 +1,20 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use chrono::{DateTime, NaiveDateTime, Utc};
 use jsonwebtoken as jwt;
 use lazy_static::lazy_static;
 use serde::Deserialize;
+use tabby_db::DbConn;
+
+use crate::schema::license::{LicenseInfo, LicenseService, LicenseStatus, LicenseType};
 
 lazy_static! {
     static ref LICENSE_DECODING_KEY: jwt::DecodingKey =
         jwt::DecodingKey::from_rsa_pem(include_bytes!("../../keys/license.key.pub")).unwrap();
 }
 
 #[derive(Debug, Deserialize)]
-#[serde(rename_all = "UPPERCASE")]
-pub enum LicenseType {
-    Team,
-}
-
-#[derive(Debug, Deserialize)]
-pub struct LicenseInfo {
+pub struct RawLicenseInfo {
```

Comment:
can we use `RawLicenseInfo` directly with some serde annotations / custom deseralizer?

Reply:
We could, but I honestly prefer to have a separate struct matching the schema of the token perfectly, and a separate function to translate it into the external representation which includes the status, similar to how we use DAOs.

Reply:
Let's use `LicenseInfo` directly. Service is allowed to relies on whatever defined in `schema`

Reply:
I think writing a custom deserializer would still be more difficult and less clear than using a struct matching the schema for deserialization followed by a conversion into another type. Would you rather I put RawLicenseInfo somewhere else and abstract the logic so it is never referred to directly outside of a utility module where it's defined?

Reply:
https://stackoverflow.com/questions/46753955/how-to-transform-fields-during-deserialization-using-serde: It looks like we can deserialize using custom functions without too much difficulty, but this isn't deserializing any data, it's using the data already deserialized in order to generate a new piece of data. That use case doesn't seem as well-supported, so we would need to write a full custom deserializer that allows us to generate the value of `status` based on the expiration date. This would be further complicated when we implement the check for `SeatsExceeded`, since that would require a database call, which I don't think we have a way of providing to the deserializer.

Reply:
For `status`, you could just use annotation `#[serde(skip)]`, and fill it in a later step?

https://serde.rs/field-attrs.html#skip